h5netcdf/core.py

```diff
@@ -621,14 +629,23 @@ def _create_child_variable(
         if shape != maxshape:
             kwargs["maxshape"] = maxshape
 
-        if isinstance(chunks, str):
-            if chunks == "h5py":
-                chunks = True
-            elif chunks == "h5netcdf":
-                chunks = _get_default_chunksizes(maxshape, dtype)
+        warn_h5py_chunking = False
+        has_unlimited_dims = 0 in shape
```

Comment:
I think this might not be true in any case (`0 in shape`). An unlimited dimension can have any (current) size, depending on already available variables. Please refer to dimension size-property. With `None in maxshape` we can be sure that an unlimited dimension is available.

Reply:
That's what this PR was originally about - the `h5py` heuristic only cares about `shape`, not `maxshape`. So manually sized unlimited dimensions actually behave like fixed dimensions (as far as chunk sizes are concerned). So the behavior is correct, but I changed the name of the variable.

Reply:
It's so easy to get lost in these intricacies. We can adjust later if necessary. Thanks.